cache/ttlmemory.go

```diff
@@ -0,0 +1,57 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+)
+
+// NewTTLInMemory creates a new thread-safe in-memory ttl cache.
+// This is most useful when storing tokens that will be handed off to
+// other services.
+func NewTTLInMemory[T any](name string) TTLCache[T] {
```

Comment:
This is more a strong opinion than an actual convention, but it's actually kind of preferable if possible to return a concrete type rather than an interface from constructors. https://duncanleung.com/go-idiom-accept-interfaces-return-types goes into some reasoning, and general experience in Evergreen is that in Go, interfaces are only really useful in a limited set of scenarios (e.g. when you must have different pluggable implementations).

Reply:
I switched it over to export it and added a simple implements subtest

Reply:
I also switched it for the otel struct